### ISSUE TITLE
staticcheck linter not reporting error when used in golangci-lint

### ISSUE THREAD

**Issue by Arvinder12** [golangci/golangci-lint#3455]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

static check not working with golangci.yaml, I have added ST1005 error in code. But that does not work. It does not report. but when i use staticcheck binary directly on the same go file, it reports correctly.

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.50.1 built from 8926a95f on 2022-10-22T10:50:47Z
```

</details>

### Configuration file

<details>

```yml
issues:
  max-issues-per-linter: 100
  max-same-issues: 100
run:
  timeout: 2m
  tests: false
  skip-dirs:
    - test/*
    - vendor/.*

linters:
  disable-all: true # for clarity, explicitly list all of the enabled linters
  enable:
    - staticcheck
  
linters-settings:
  staticcheck:
    go: 1.18
    checks: ["all", "ST1005"]
```

</details>

### Go environment

<details>

```console
$ go version
go version go1.18 linux/amd64
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /mnt/c/Users/t0276614/go/src/practice /mnt/c/Users/t0276614/go/src /mnt/c/Users/t0276614/go /mnt/c/Users/t0276614 /mnt/c/Users /mnt/c /mnt / /root]
INFO [config_reader] Used config file .golangci.yaml
INFO [lintersdb] Active 1 linters: [staticcheck]
INFO [loader] Go packages loading at mode 575 (types_sizes|compiled_files|exports_file|imports|name|deps|files) took 286.657699ms
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 5.787576ms
INFO [linters_context/goanalysis] analyzers took 868.1622ms with top 10 stages: buildir: 694.588616ms, fact_deprecated: 46.321707ms, SA4018: 32.859762ms, SA5012: 25.195503ms, nilness: 25.032332ms, fact_purity: 20.372926ms, typedness: 17.702836ms, isgenerated: 3.749662ms, SA9006: 463.739µs, SA4027: 158.363µs
INFO [runner] processing took 4.427µs with stages: max_same_issues: 736ns, cgo: 548ns, skip_dirs: 478ns, nolint: 466ns, max_from_linter: 352ns, filename_unadjuster: 193ns, autogenerated_exclude: 193ns, path_prettifier: 192ns, uniq_by_line: 192ns, exclude: 187ns, skip_files: 181ns, sort_results: 88ns, source_code: 86ns, diff: 82ns, exclude-rules: 78ns, identifier_marker: 77ns, severity-rules: 76ns, path_shortener: 75ns, max_per_file_from_linter: 74ns, path_prefixer: 73ns
INFO [runner] linters took 763.95696ms with stages: staticcheck: 763.901786ms
INFO File cache stats: 0 entries of total size 0B
INFO Memory: 12 samples, avg is 108.2MB, max is 173.0MB
INFO Execution took 1.06598375s
```

</details>

### Code example or link to a public repository

<details>

```go
var ErrInvalidLogLevelType = errors.New("Invalid log level type")
```

</details>

**Comment by ldez** [comment]
Hello,

`staticcheck` is a binary **and** a set of rules, those rules are grouped into "category": `staticcheck`, `gosimple`, `stylecheck` and `unused`.

- [staticcheck](https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/staticcheck/analysis.go#L23)
- [stylecheck](https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/stylecheck/analysis.go#L14)
- [simple](https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/simple/analysis.go#L11)
- [unused](https://github.com/dominikh/go-tools/blob/331bcca40621580531c9b8a9cb7d5c2c06c709d2/unused/unused.go#L418)

Inside golangci-lint we use the sets of rules, not the binary.

The rule `ST1005` is related to the `stylecheck` set:
- `STxxx` -> `stylecheck`
- `SAxxx` -> `staticcheck`

```yaml
issues:
  max-issues-per-linter: 100
  max-same-issues: 100

run:
  timeout: 2m
  tests: false
  skip-dirs:
    - test/*
    - vendor/.*
  go: '1.18'

linters:
  disable-all: true # for clarity, explicitly list all of the enabled linters
  enable:
    - stylecheck
```

```console
$ golangci-lint run
foo.go:8:30: ST1005: error strings should not be capitalized (stylecheck)
var ErrInvalidLogLevelType = errors.New("Invalid log level type")
                             ^
```

**Comment by Arvinder12** [comment]
Thank you for the information
